src/Uno.UWP/Storage/StorageFile.cs

```diff
@@ -90,142 +194,117 @@ public async Task<Stream> GetThumbnailAsync(CancellationToken ct, ThumbnailMode
+			};
 
-		public async Task MoveAsync(CancellationToken ct, StorageFolder targetFolder)
-		{
-			await MoveAsync(ct, targetFolder, Name, NameCollisionOption.FailIfExists);
+			return await destinationFolder.CreateFileAsync(desiredNewName, creationOption).AsTask(ct);
 		}
 
-		public async Task MoveAsync(CancellationToken ct, StorageFolder targetFolder, string desiredNewName)
-		{
-			await MoveAsync(ct, targetFolder, desiredNewName, NameCollisionOption.FailIfExists);
-		}
-
-		public async Task MoveAsync(CancellationToken ct, StorageFolder targetFolder, string desiredNewName, NameCollisionOption option)
-		{
-			// TODO: Check the _scheme of the target folder
-			var targetPath = global::System.IO.Path.Combine(targetFolder.Path, desiredNewName);
-
-			if (File.Exists(targetPath))
+		private static string GetContentTypeFromFileType(string fileType)
+			// https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+			=> fileType.ToLowerInvariant() switch
 			{
-				switch (option)
-				{
-					case NameCollisionOption.FailIfExists:
-						throw new IOException("File {0} already exists".InvariantCultureFormat(targetPath));
-
-					case NameCollisionOption.GenerateUniqueName:
-						var extension = global::System.IO.Path.GetExtension(desiredNewName);
-						desiredNewName = global::System.IO.Path.ChangeExtension(desiredNewName, Guid.NewGuid().ToString("N", CultureInfo.InvariantCulture) + extension);
-						await MoveAsync(ct, targetFolder, desiredNewName, option);
-						return;
-
-					case NameCollisionOption.ReplaceExisting:
-						File.Delete(targetPath);
-						break;
-
-					default:
-						throw new ArgumentOutOfRangeException("option");
-				}
-			}
-
-			switch (Scheme)
-			{
-				default:
-					File.Move(Path, targetPath);
-					break;
-			}
-
-			_fileUri = new Uri("file://" + targetPath);
-		}
-
-		public IAsyncOperation<BasicProperties> GetBasicPropertiesAsync()
-			=> GetBasicPropertiesAsync(new CancellationToken()).AsAsyncOperation<BasicProperties>();
-
-		public async Task<BasicProperties> GetBasicPropertiesAsync(CancellationToken ct)
-		{
-			return new BasicProperties(this);
-		}
-
-		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
-		public static global::Windows.Foundation.IAsyncOperation<global::Windows.Storage.StorageFile> GetFileFromApplicationUriAsync( global::System.Uri uri)
-		{
-			return AsyncOperation.FromTask(ct => GetFileFromApplicationUriAsyncTask(ct, uri));
-		}
-
-#if NET461 || __SKIA__ || __NETSTD_REFERENCE__
-		private static Task<StorageFile> GetFileFromApplicationUriAsyncTask(CancellationToken ct, Uri uri)
-		{
-			throw new NotImplementedException();
-		}
-#endif
+				".aac" => "audio/aac",
+				".abw" => "application/x-abiword",
+				".arc" => "application/x-freearc",
+				".avi" => "video/x-msvideo",
+				".azw" => "application/vnd.amazon.ebook",

```

Comment:
This should be put in a kind of service somewhere... it's usually something managed by the OS. The application may want to extend this.

Reply:
It is service :)
See MimeTypeMap.Singleton.GetMimeTypeFromExtension (at least, on Android)